bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandlerFactory.java

```diff
@@ -0,0 +1,374 @@
+import org.openhab.binding.magentatv.internal.utils.MagentaTVException;
+import org.openhab.binding.magentatv.internal.utils.MagentaTVLogger;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+
+/**
+ * The {@link MagentaTVHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { ThingHandlerFactory.class, MagentaTVHandlerFactory.class }, configurationPid = "binding."
+        + BINDING_ID)
+public class MagentaTVHandlerFactory extends BaseThingHandlerFactory {
+    private final MagentaTVLogger logger = new MagentaTVLogger(MagentaTVHandlerFactory.class, "Factory");
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_RECEIVER);
+
+    private final MagentaTVNetwork network = new MagentaTVNetwork();;
+    private @Nullable MagentaTVPoweroffListener upnpListener;
+    private boolean servletInitialized = false;
+
+    @SuppressWarnings("null")
+    @NonNullByDefault
+    protected class MagentaTVDevice {
+        protected String udn = "";
+        protected String mac = "";
+        protected String deviceId = "";
+        protected String ipAddress = "";
+        protected Map<String, Object> properties = new HashMap<String, Object>();
+        protected @Nullable MagentaTVHandler thingHandler;
+    }
+
+    private final Map<String, MagentaTVDevice> deviceList = new HashMap<String, MagentaTVDevice>();
+
+    /**
+     * Activate the bundle: save properties
+     *
+     * @param componentContext
+     * @param configProperties set of properties from cfg (use same names as in
+     *            thing config)
+     */
+
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> configProperties) {
+        super.activate(componentContext);
+        logger.debug("Activate HandlerFactory");
+    }
+
+    /**
+     * return list of supported thins
+     */
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        logger.debug("Create thing type {0}", thing.getThingTypeUID().getAsString());
+        if (THING_TYPE_RECEIVER.equals(thingTypeUID)) {
+            try {
+                return new MagentaTVHandler(this, thing, network);
+            } catch (RuntimeException e) {
+                logger.exception(e, "Network initialization failed");
```

Comment:
Reduce to DEBuG

Reply:
strange situation, inform user
message corrected

Reply:
see above